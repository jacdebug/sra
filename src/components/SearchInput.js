import React from 'react';
import styled from 'styled-components';

const SearchInput = ({ searchTerm, searchKeyPress }) =>
  <input
    type="text"
    placeholder="Seach by file name... "
    value={searchTerm}
    onKeyPress={searchKeyPress}
    onChange={searchKeyPress}
  />;

export const StyledInput = styled.input`padding: 5px;`;
export default SearchInput;
