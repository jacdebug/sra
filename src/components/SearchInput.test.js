import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchInput from './SearchInput';

it('renders without crashing', () => {
  const mockFn = jest.fn();
  shallow(<SearchInput searchTerm="testval" searchKeyPress={mockFn} />);
});

it('renders correctly', () => {
  const mockFn = jest.fn();
  const tree = shallow(
    <SearchInput searchTerm="testval" searchKeyPress={mockFn} />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
