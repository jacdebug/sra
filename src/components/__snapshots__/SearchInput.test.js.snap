// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Seach by file name... "
  type="search"
  value="testval"
/>
`;
